include/net/cloud.h

```diff
@@ -112,19 +112,15 @@ struct cloud_event {
 	} data;
 };
 
-/**
- * @brief Cloud last will and testament type. See:
- * http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html
- * sections  3.1.2.5-3.1.2.7, 3.1.3.2-3.1.3.3.
- */
-struct cloud_lwt {
-	u8_t qos;		/**< MQTT QoS value. */
-	/* Broker to retain will for new clients.  Not currently
-	 * supported by AWS IoT.
-	 */
-	bool retain;
-	const char *topic;	/**< If specified, @ref message must also. */
-	const char *message;	/**< Message sent to topic by broker. */
+/**@brief Cloud configuration. */
+struct cloud_cfg {
```

Comment:
I was thinking that we'd put this into `struct cloud_backend_config`. Then it can be accessed anywhere by using `device_get_binding(<backend name>)->mqtt...`. Thoughts, @rlubos ?

Reply:
I believe you meant `cloud_get_binding`, not `device_get_binding`?

This sounds quite reasonable, given the fact that there already is `struct cloud_backend_config`. Having `struct cloud_cfg` could be confusing.

To be honest, there's no really need for any specifics in the generic cloud API. `struct cloud_backend_config` could simply hold a void pointer to some opaque, backend-specific structure, and the actual cloud implementations could make use of it (i. e. initialize the pointer with backend-specific data and make use of it when needed).

Reply:
You're right, `cloud_get_binding()` is what it should say.

And yeah, a void pointer would also do the trick. There already exists a `void *user_data` in the backend config that's piggybacked to all events. An additional one for custom data to the backend would fit in.

Reply:
I hadn't noticed these comments on Apr 23.  I just updated the PR locally to build with 1.3.0.  I'll address the comments above soon.